.github/workflows/format_and_lint_ui.yml

```diff
@@ -45,7 +45,7 @@ jobs:
       - name: Install dependencies
         run: |
           pip install -e .
-          npm install --prefix ./ui
+          npm install --prefix ./ui --legacy-peer-deps
```

Comment:
`--legacy-peer-deps` becaus the node version is old ?

Reply:
It's just a flag for NPM to disregard peer deps warnings and install any missing peer deps automatically. With `pnpm`, I would have added an override in `package.json`. The warning just comes from `@testing-library/cypress` which does not allow the latest version of Cypress as peer dep — no problem with ignoring this warning if the tests pass.